lib/elixir/lib/module.ex

```diff
@@ -1097,14 +1106,120 @@ defmodule Module do
         :ok
       {:error, :private_doc} ->
         :elixir_errors.warn line, env.file,
-          "function #{name}/#{arity} is private, " <>
+          "function #{format_function(pair)} is private, " <>
           "@doc's are always discarded for private functions"
     end
 
     :ok
   end
 
   @doc false
+  # Used internally to check impl usage.
+  # This function is private and must be used only internally.
+  def check_impl(env, kind, name, args, _guards, _body) do
+    module = env.module
+    table = data_table_for(module)
+
+    value = get_impl_info(table)
+
+    line = env.line
+    file = env.file
```

Comment:
Please use pattern matching to extract line, file, module and others out.

    %{module: module, line: line, file: file} = env